File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res=INT_MIN, sum=0, n=nums.size();
        for(int i=0;i<n;i++){
            sum+=nums[i];
            res=max(res, sum);
            if(sum<0)   sum=0;
        }
        return res;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = float("-inf")
        temp = 0
        for i in nums:
            temp += i
            res = max(res, temp)
            if temp < 0:
                temp = 0
        return res